### PR TITLE
[2/N] Fix clang-tidy warnings in aten/src/ATen/native/*.{cpp,h} 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #130798

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/BatchLinearAlgebraKernel.cpp [hunk 1]
```diff
@@ -352,7 +352,7 @@ static void apply_geqrf(const Tensor& input, const Tensor& tau) {
 
   // if lwork is less than 'n' then a warning is printed:
   // Intel MKL ERROR: Parameter 7 was incorrect on entry to SGEQRF.
-  lwork = std::max<int>(std::max<int>(1, n), real_impl<scalar_t, value_t>(wkopt));
+  lwork = std::max<int>(std::max<int64_t>(1, n), real_impl<scalar_t, value_t>(wkopt));
```
Comment (Skylion007) [inline aten/src/ATen/native/BatchLinearAlgebraKernel.cpp:355]: Hmm, still doing a narrow here with `std::max`

File: aten/src/ATen/native/Pool.h [hunk 2]
```diff
@@ -120,7 +120,7 @@ inline std::pair<c10::SymInt, c10::SymInt> pooling_same_mode_padding_lr(
 inline void
 pool2d_shape_check(
   const Tensor& input,
-  int kH, int kW, int dH, int dW, int padH, int padW, int dilationH, int dilationW,
+  int64_t kH, int64_t kW, int64_t dH, int64_t dW, int64_t padH, int64_t padW, int64_t dilationH, int64_t dilationW,
```
Comment (Skylion007) [inline aten/src/ATen/native/Pool.h:123]: Why does dilationH, dilationW, etc need to be int_64?

File: aten/src/ATen/native/Convolution.cpp [hunk 3]
```diff
@@ -818,7 +818,7 @@ static auto view3d(const at::Tensor& tensor) -> at::Tensor {
   return tensor.squeeze(2);
 }
 
-static at::Tensor subtensor(at::Tensor& tensor, int dim, int groups, int g) {
+static at::Tensor subtensor(at::Tensor& tensor, int64_t dim, int64_t groups, int64_t g) {
```
Comment (Skylion007) [inline aten/src/ATen/native/Convolution.cpp:821]: We support int64 sized groups? Our dim count is hardcoded way lower than int32max too

File: aten/src/ATen/native/Pool.h (hunk 2 above)
Comment (cyyever) [reply aten/src/ATen/native/Pool.h:123]: Because the call sites pass int_64 to them

File: aten/src/ATen/native/Convolution.cpp (hunk 3 above)
Comment (cyyever) [reply aten/src/ATen/native/Convolution.cpp:821]: The same reason, the call sites pass int64

File: aten/src/ATen/native/Convolution.cpp [hunk 4]
```diff
@@ -1896,7 +1896,7 @@ std::tuple<Tensor,Tensor,Tensor> _convolution_double_backward( const std::option
         auto i_size = input.sizes();
         for (const auto i : c10::irange(2, gI_size.size())) {
```
Comment (Skylion007) [inline aten/src/ATen/native/Convolution.cpp:1897]: Do we not have a utility for an int64 c10::range? Seems like an oversight if so.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**